lib/src/wiggle_abi/dictionary_impl.rs

```diff
@@ -6,23 +6,62 @@ use {
         session::Session,
         wiggle_abi::{fastly_dictionary::FastlyDictionary, types::DictionaryHandle},
     },
+    std::{
+        convert::{
+            TryFrom,
+            TryInto,
+        },
+        fs,
+        path::Path
+    },
     wiggle::GuestPtr,
 };
 
+fn read_json_file<P: AsRef<Path>>(file: P) -> serde_json::Map<String, serde_json::Value> {
+    let data = fs::read_to_string(file).expect("Unable to read file");
+    let json: serde_json::Value = serde_json::from_str(&data).expect("JSON was not well-formatted");
+    let obj = json.as_object().expect("Expected the JSON to be an Object");
+    obj.clone()
+}
+
 impl FastlyDictionary for Session {
-    #[allow(unused_variables)] // FIXME: Remove this directive once implemented.
     fn open(&mut self, name: &GuestPtr<str>) -> Result<DictionaryHandle, Error> {
-        Err(Error::NotAvailable("Dictionary lookup"))
+        let name = name.as_str()?.to_owned();
+        self.dictionary_handle(&name)
```

Comment:
FWIW, should be able to avoid copying the name like so:

```suggestion
        self.dictionary_handle(&name.as_str()?)
```

Putting the `&` in front allows Rust to trigger "deref coercion" and turn a [`GuestStr`](https://docs.rs/wiggle/0.29.0/wiggle/struct.GuestStr.html) into a `&str`.

Reply:
That's really nice ☺️ This change was done in [`84661a9` (#61)](https://github.com/fastly/Viceroy/pull/61/commits/84661a9df860000c08fc1e73035c2999edc8b5d3)